backend/src/api/books/books.controller.ts

```diff
@@ -0,0 +1,54 @@
+import { BadRequestException, Body, Controller, Get, Post, Query } from '@nestjs/common'
+import { ApiTags, ApiQuery } from '@nestjs/swagger'
+
+import { CreateBookDto } from './dto/createBook.dto'
+import { BooksService } from './books.service'
+
+@ApiTags('books')
+@Controller('/api/books')
+export class BooksController {
+  constructor(private booksService: BooksService) {}
+
+  @Get('/')
+  @ApiQuery({ name: 'limit', required: false, type: Number })
+  @ApiQuery({ name: 'page', required: false, type: Number })
+  async get(@Query('limit') limit: number = 6, @Query('page') page: number = 1): Promise<any> {
+    try {
+      return {
+        message: 'Get New Books Successfully',
+        data: await this.booksService.find(limit, page),
+      }
+    } catch (error) {
+      throw new BadRequestException(error.message)
+    }
+  }
+
+  @Post()
+  async create(@Body() createBookDto: CreateBookDto): Promise<any> {
```

Comment:
a Books will have relation with authors, manufacturer, types of book. Please re check this API

Reply:
I added createdBy field so that we know the person create the book. Besides the author should be a string so that we can modify it and it's not relationship to another table